crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -420,25 +417,20 @@ impl<'w> WorldChildBuilder<'w> {
 
     /// Spawns an [`Entity`] with no components and inserts it into the children defined by the [`WorldChildBuilder`] which adds the [`Parent`] component to it.
     pub fn spawn_empty(&mut self) -> EntityMut<'_> {
-        let parent_entity = self.parent_entity();
-        let entity = self.world.spawn(Parent(parent_entity)).id();
-        push_child_unchecked(self.world, parent_entity, entity);
-        self.current_entity = Some(entity);
+        let entity = self.world.spawn(Parent(self.parent)).id();
+        push_child_unchecked(self.world, self.parent, entity);
         if let Some(mut added) = self.world.get_resource_mut::<Events<HierarchyEvent>>() {
```

Comment:
Are event's resources?

Why not just send the event, isn't a child always being added?

Reply:
The if let here is checking if the event resource itself exists (to reduce random panics).